pkg/kubectl/cmd/scale.go

```diff
@@ -153,6 +155,21 @@ func RunScale(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []stri
 			errs = append(errs, err)
 			continue
 		}
+		if cmdutil.GetRecordFlag(cmd) {
+			patchBytes, err := cmdutil.ChangeResourcePatch(info, f.Command())
```

Comment:
Although RecordChangeSource just adds the annotation. I think what you have here is fine.